### PR TITLE
Previous QSL Check

### PR DESCRIPTION
**Pull request by HB9HIL**
image

Show an additional column in the QSL queue which shows if there were already sent QSL cards on the same band/mode (or if not empty sat_name) to that station.

### CODE REVIEW

File: application/models/Qslprint_model.php [hunk 1]
```diff
@@ -159,6 +159,29 @@ function show_oqrs($id) {
...
+			$where[] = "(COL_CALL = ? AND COL_MODE = ? AND COL_BAND = ? AND COL_SAT_NAME = ?)";
+			$binding[] = $qso['call'];
+			$binding[] = $qso['mode'];
+			$binding[] = $qso['band'];
+			$binding[] = $qso['sat_name'] ?? '';
+		}
+	
+		$sql = "SELECT COL_CALL, COL_MODE, COL_BAND, COL_SAT_NAME, COUNT(COL_PRIMARY_KEY) AS count FROM " . $this->config->item('table_name') . " 
+			WHERE COL_QSL_SENT = 'Y'
+			AND (" . implode(' OR ', $where) . ")
```
Comment (int2001) [inline application/models/Qslprint_model.php:179]: station_id is missing. this one checks every QSO in the instance

**Review by int2001 — CHANGES_REQUESTED** [review]

**Comment by AndreasK79** [comment]
@HB9HIL you must fix when lines are added to the table as well:
image
Click on QSO list, and add a QSO. It will then look like the picture.

**Review by int2001 — APPROVED** [review]
works now as expected and shows the amount of already sent QSLs for:
- same call
- on same band and mode
- on same SAT (if SAT-QSO)
- out of same station-location

**Comment by AndreasK79** [comment]
Javascript still needs to be fixed when adding lines.

**Comment by HB9HIL** [comment]
> Javascript still needs to be fixed when adding lines (just a reminder).

yep I know. working on it later that day

**Comment by HB9HIL** [comment]
> Javascript still needs to be fixed when adding lines (just a reminder).

fixed

**Comment by AndreasK79** [comment]
Errors gone, merge time :)

**Review by AndreasK79 — APPROVED** [review]

**Pull request merged**